src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -354,13 +354,24 @@ PAL_TlsIo AppleCryptoNative_SslRead(SSLContextRef sslContext, uint8_t* buf, uint
     return OSStatusToPAL_TlsIo(status);
 }
 
-int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore)
+int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore, int32_t* pOSStatus)
 {
+    if (pOSStatus == NULL)
+        return -1;
+
     if (sslContext == NULL || notBefore == NULL)
+    {
+        *pOSStatus = errSecInvalidData;
```

Comment:
This is not following the conventions of the shim.

Don't set *pOSStatus to anything other than the OSStatus value.  The negative returns are for diagnosing incorrect calls,  They'll throw exceptions already.

Reply:
errSecInvalidData should be one of OSStatus values. I could take it out since @stephentoub comment on it as well. The idea was that we would have better message in trace log.  Since the outcome is different it would be still claer what happened IMHO.

Reply:
Except it won't be in the log.  Returning a value other than 0 or 1 will throw long before your log statement.

Assigning errSecMemoryError is OK (except doesn't matter since it's only set in a negative-return path) to acknowledge that a create call failed, but otherwise it should only be assigned based on platform calls.  The distinction is "negative return" means a shim error, "returned 0 and there's an osStatus value" means "runtime failure, the OS told us why".